### PR TITLE
Add Search box to Report Source Selection

### PR DESCRIPTION
**Pull request by davidcui1225**
*Issue #, if available:*
#264 
*Description of changes:*
* Changed `Report source` selection in `Report settings` to be `EuiComboBox` vs. Drop-down list. Now users can search for their report sources.
* `Create report definition` no longer has pre-populated report source, users will have to enter it manually
* Added front-end input validation for Report source to verify field is not empty
* Updated cypress test to select report source

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by davidcui1225** [comment]
codecov failed- will add more tests

**Review by akbhatta — APPROVED** [review]

File: kibana-reports/public/components/report_definitions/report_settings/__tests__/report_settings_helpers.test.tsx [hunk 1]
```diff
@@ -13,6 +13,7 @@
  * permissions and limitations under the License.
  */
 
+import { exception } from 'console';
```
Comment (zhongnansu) [inline kibana-reports/public/components/report_definitions/report_settings/__tests__/report_settings_helpers.test.tsx]: I didn't see this is used anywhere, can you remove?

**Review by zhongnansu — COMMENTED** [review]
Anything else you can do to resolve the codecov failure?

File: kibana-reports/public/components/report_definitions/report_settings/__tests__/report_settings_helpers.test.tsx (hunk 1 above)
Comment (davidcui1225) [reply kibana-reports/public/components/report_definitions/report_settings/__tests__/report_settings_helpers.test.tsx]: Good catch- must have been added when I selected wrong drop-down option in tests, will remove

**Comment by davidcui1225** [comment]
> codecov failed- will add more tests

Codecov fixed- coverage raised significantly

**Review by zhongnansu — APPROVED** [review]

**Pull request merged**